server-plugins/notification-resources/src/index.ts

```diff
@@ -801,24 +810,24 @@ async function updateContextsTimestamp (
     }
   }
 
-  await control.apply(res)
+  if (res.length > 0) {
+    await ctx.with('apply', {}, () => control.apply(res))
+  }
 }
 
 async function removeContexts (
+  ctx: MeasureContext,
   contexts: DocNotifyContext[],
   unsubscribe: Ref<PersonAccount>[],
   control: TriggerControl
 ): Promise<void> {
   if (contexts.length === 0) return
   if (unsubscribe.length === 0) return
 
-  const unsubscribeAccounts = await control.modelDb.findAll(contact.class.PersonAccount, {
-    _id: { $in: unsubscribe }
-  })
   const res: Tx[] = []
 
   for (const context of contexts) {
-    const account = unsubscribeAccounts.find(({ _id }) => _id === context.user)
```

Comment:
We should add something like to remove only correct contexts: 
```
if(!unsubscribe.includes(context.user)) {
continue;
}
```

Reply:
Fixed.